azext_iot/common/base_discovery.py

```diff
@@ -178,9 +178,18 @@ def find_resource(self, resource_name: str, rg: str = None):
 
         resource_list = self.get_resources()
 
+        for resource in resource_list:
+            if resource_name.lower() == resource.name.lower():
+                target = resource
+
         if resource_list:
+            is_dict = isinstance(resource_list[0], dict)
```

Comment:
Still supporting both model & modeless based sdk

Reply:
I swear that I thought I removed it